compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -224,6 +227,46 @@ internal class BindingModuleGenerator(
     }
   }
 
+  private fun checkNotGeneric(
+    type: ClassDescriptor,
+    boundTypeDescriptor: ClassDescriptor
+  ) {
+    if (boundTypeDescriptor.declaredTypeParameters.isNotEmpty()) {
+
+      fun StringBuilder.describeTypeParameters(type: KotlinType): StringBuilder {
+        if (type.arguments.isNotEmpty()) {
+          append("<")
+
+          for ((count, typeArgument) in type.arguments.withIndex()) {
+            if (count > 0) {
+              append(", ")
+            }
+            append(DescriptorUtils.getClassDescriptorForType(typeArgument.type).name)
+            describeTypeParameters(typeArgument.type)
+          }
+
+          append(">")
+        }
+
+        return this
+      }
```

Comment:
This entire function could be shrinked to:
```kotlin
      fun KotlinType.describeTypeParameters(): String =
        arguments
            .ifEmpty { return "" }
            .joinToString(prefix = "<", postfix = ">") { typeArgument ->
              typeArgument.type.classDescriptorForType().name.asString() +
                  typeArgument.type.describeTypeParameters()
            }
```
You're using a `StringBuilder`, but I don't think this optimization is worth it in an error scenario. I'd prefer better readability in this case.

Reply:
Very fancy! Love it. Thanks!